packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -449,18 +480,36 @@ class SelectableRegionState extends State<SelectableRegion> with TextSelectionDe
     );
   }
 
-  void _startNewMouseSelectionGesture(DragDownDetails details) {
-    widget.focusNode.requestFocus();
-    hideToolbar();
-    _clearSelection();
+  void _startNewMouseSelectionGesture(TapDragDownDetails details) {
+    switch (_getEffectiveConsecutiveTapCount(details.consecutiveTapCount)) {
+      case 1:
+        widget.focusNode.requestFocus();
+        hideToolbar();
+        _clearSelection();
+      case 2:
+        _selectWordAt(offset: details.globalPosition);
+    }
   }
 
-  void _handleMouseDragStart(DragStartDetails details) {
-    _selectStartTo(offset: details.globalPosition);
+  void _handleMouseDragStart(TapDragStartDetails details) {
+    switch (_getEffectiveConsecutiveTapCount(details.consecutiveTapCount)) {
+      case 1:
+        _selectStartTo(offset: details.globalPosition);
+    }
   }
 
-  void _handleMouseDragUpdate(DragUpdateDetails details) {
-    _selectEndTo(offset: details.globalPosition, continuous: true);
+  void _handleMouseDragUpdate(TapDragUpdateDetails details) {
+    switch (_getEffectiveConsecutiveTapCount(details.consecutiveTapCount)) {
+      case 1:
+        _selectEndTo(offset: details.globalPosition, continuous: true);
+      case 2:
+        _selectEndTo(offset: details.globalPosition, continuous: true, textGranularity: TextGranularity.word);
```

Comment:
Which class is synthesizing the `start` event to selectable?

Reply:
`_startNewMouseSelectionGesture` is the one setting the `start` edge with `_selectWordAt`, or were you referring to something else?

Reply:
I am referring to cases where the `_selectEndTo` reach a new selectable other than the `_selectWordAt` originally sent to

Reply:
`_SelectableRegionContainerDelegate` is synthesizing the first start event on `ensureChildUpdated` when `_selectEndTo` reaches a new selectable.